src/index.rs

```diff
@@ -128,6 +128,25 @@ impl<T: Indexable, N: ArrayLength<T>> Index<T, N> {
         (index as usize, dist)
     }
 
+    pub fn find_nearest_neighbors(&self, point: &Vec<Datum<T, N>>, num:usize) -> (Vec<i32>, Vec<T::ResultType>) {
```

Comment:
I'll fix the return value to not return more items than present in the index, and the index type will be `usize` rather than `i32`.